### PR TITLE
feat(main): #1527 added help page route and remove the dialog

### LINKED ISSUE #1527: Move `help` dialog to have a page with url string, for deep linking
Move `help` dialog to have a page with url string, for deep linking.

Sometimes, I want to share a link to the help page(s), but as its a dialog it is not possible. We should move to a help page, may be something like...

```
/help/{path}
```

```typescript
public topics: Help[] = [
    {
      title: 'Quickstart',
      description: 'Overview information on getting started',
      icon: 'quickstart_icon',
      path: 'quickstart',
      updatedAt: '09/13/2019 09:05 AM',
    },
];
```

### PR DESCRIPTION
**Pull request by webkhushboo**
closes #1527 

### Notes

A summary of what was achieved in this PR

- [ ] Removed the help dialog
- [ ] Added the route for help page when click on help title 

image

### CODE REVIEW

File: web/src/app/shared/dialog/markdown/markdown.component.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { Help } from '@app/main/components/help/help';
+import { HelpModel } from '@shared/models/index.model';
+
+@Component({
+  selector: 'dashboard-markdown',
+  templateUrl: './markdown.component.html',
+  styleUrls: ['./markdown.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class MarkdownComponent implements OnInit {
```
Comment (eddiejaoude) [inline web/src/app/shared/dialog/markdown/markdown.component.ts]: `MarkdownComponent` implies it is a generic component but it is specific to `help`, `HelpComponent` I think would make more sense?

File: web/src/app/shared/models/help.model.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  constructor() {
+    this.content = '';
+    this.icon = '';
+    this.content = '';
+  }
+
+  /**
+   * Returns the help topics list
+   */
+  getTopicsList(): Help[] {
```
Comment (eddiejaoude) [inline web/src/app/shared/models/help.model.ts]: I don't think this is needed, the `topics` property is `public`

Comment (webkhushboo) [reply web/src/app/shared/models/help.model.ts]: okay will remove

File: web/src/app/shared/dialog/markdown/markdown.component.ts (hunk 1 above)
Comment (webkhushboo) [reply web/src/app/shared/dialog/markdown/markdown.component.ts]: HelpComponent is already there . What should be the name for this component ?
I have just renamed the dialog-markdown.component to markdown.component

Comment (eddiejaoude) [reply web/src/app/shared/dialog/markdown/markdown.component.ts]: May be, `HelpDetailComponent`?

Comment (webkhushboo) [reply web/src/app/shared/dialog/markdown/markdown.component.ts]: Done

File: web/src/app/main/components/help/help.component.ts [hunk 3]
```diff
@@ -19,65 +17,16 @@ export class HelpComponent implements OnInit {
...
     private http: HttpClient,
-    private form: FormBuilder,
-    private dialog: MatDialog
+    private form: FormBuilder
   ) { }
 
   ngOnInit(): void {
+    this.topics = this.help.topics;
     this.topics.forEach((help: { title: string, path: string }, index: number) => {
       this.http.get(`/assets/help/${help.path}.md`, { responseType: 'text' }).subscribe((content: string) => {
```
Comment (eddiejaoude) [inline web/src/app/main/components/help/help.component.ts]: I know this was there already, but might be cleaner to do it as a resolve?

File: web/src/app/shared/models/help.model.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { Help } from '@app/main/components/help/help';
+
+export class HelpModel {
+  icon: string;
+  title: string;
+  content: string;
+  topics: Help[] = [
```
Comment (eddiejaoude) [inline web/src/app/shared/models/help.model.ts]: I know this was here before, but `Help` should be a model, may be `HelpTopic`?

Comment (webkhushboo) [reply web/src/app/shared/models/help.model.ts]: okay

File: web/src/app/main/components/help/help.component.ts (hunk 3 above)
Comment (webkhushboo) [reply web/src/app/main/components/help/help.component.ts]: okay

**Comment by eddiejaoude** [comment]
All looks good. What did you think of the resolve idea?

File: web/src/app/core/resolvers/help.resolver.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  ) { }
+
+  /**
+   * Finds the help content in resolver
+   * @param route ActivatedRouteSnapshot
+   */
+  resolve(route: ActivatedRouteSnapshot): Observable<String> {
+    const path: string = route.params.path;
+
+    return this.http.get(`/assets/help/${path}.md`, { responseType: 'text' }).pipe(
```
Comment (eddiejaoude) [inline web/src/app/core/resolvers/help.resolver.ts]: `pipe` usually goes on a new line. I can update

File: web/src/app/main/components/help-detail/help-detail.component.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+  public data: HelpModel = new HelpModel();
+
+  /**
+   * Life cycle method
+   * @param http HttpClient instance
+   * @param route ActivatedRoute instance
+   */
+  constructor(
+    private http: HttpClient,
```
Comment (eddiejaoude) [inline web/src/app/main/components/help-detail/help-detail.component.ts]: `HttpClient` is not used anymore, I can update

File: web/src/app/main/components/help/help.component.html [hunk 7]
```diff
@@ -19,7 +19,7 @@
     <mat-card-content class="help__card__content" fxLayout="column" fxLayoutGap="20px">
       <mat-divider></mat-divider>
       <mat-nav-list *ngFor="let help of filteredTopics">
-        <a (click)="openDialog(help)">
+        <a [routerLink]="[help.path]">
```
Comment (eddiejaoude) [inline web/src/app/main/components/help/help.component.html]: I think full path is preferred, I can update

File: web/src/app/shared/models/help.model.ts [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    },
+  ];
+
+  /**
+   * Life cycle method
+   */
+  constructor() {
+    this.content = '';
+    this.icon = '';
+    this.content = '';
```
Comment (eddiejaoude) [inline web/src/app/shared/models/help.model.ts]: I don't think these defaults are needed? I can update

**Comment by eddiejaoude** [comment]
Please review my commit https://github.com/DashboardHub/PipelineDashboard/pull/1529/commits/2b0d88d373e528e45e2773a4ae572798be4f384d

**Pull request merged**